src/app.ts

```diff
@@ -13,14 +13,85 @@ interface TemperatureSummary {
   average: number
 }
 
+interface daySumary {
+  time: string
+  temperature: number
+  city: string
+  arrivalNumber: number
+}
+
+const listDaySumary: daySumary[][] = [];
+let sumaryOfDay: daySumary[] = [];
+let tempReading: TemperatureReading;
+let order: number;
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  tempReading = readings[0];
+  order = 1;
+  readings.forEach((dailyTemperature) => {
+    if (
+      dailyTemperature.time.toLocaleDateString() ===
+        tempReading.time.toLocaleDateString() &&
+      dailyTemperature.city === tempReading.city
+    ) {
+      sumaryOfDay.push({
+        time: dailyTemperature.time.toLocaleDateString(),
```

Comment:
If the city and the day always be the same in `summaryOfDay` register, why repeat in each register the same data?